spec/README.md

```diff
@@ -0,0 +1,85 @@
+# Driver Tests
+
+The tests run against a MongoDB cluster which is
+configured and started externally to the test suite. This allows
+running the entire test suite against, for example, a standalone
+mongod as well as a replica set. The flip side to this is the
+test suite will not work without a running mongo cluster, and
+tests which are not applicable to or cannot be performed on the
+running mongo cluster are skipped.
+
+Not only does the test suite require an externally launched cluster,
+the test suite must also be told how the cluster is configured
+via MONGODB_URI, TOPOLOGY, MONGODB_ADDRESSES, RS_ENABLED, RS_NAME and
```

Comment:
I think it's worth documenting that the user needs to set _either_ `MONGODB_URI` or a valid set of `MONGODB_ADDRESSES`, `RS_ENABLED`, `RS_NAME`, and `SHARDED_ENABLED`.

Reply:
Not true - they don't need to set any of them.

Reply:
Apologies, I worded that poorly; my point is that if a user sets `MONGODB_URI`, they don't need to set the others (and vice-versa), and using "and" here is a bit misleading.

Reply:
Also not true, and this discussion is outside the scope of this PR anyway.

Reply:
Have it your way.